DQMOffline/Trigger/plugins/BPHMonitor.cc

```diff
@@ -364,13 +374,40 @@ void BPHMonitor::analyze(edm::Event const& iEvent, edm::EventSetup const& iSetup
 
   edm::Handle<reco::BeamSpot> beamSpot;
   iEvent.getByToken( bsToken_,  beamSpot);
+  if ( !beamSpot.isValid() ) {
+    if (!warningPrinted4token_[0]) {
+      edm::LogWarning("BPHMonitor") << "skipping events because the collection " << bsInputTag_.label().c_str() << " is not available";
+      warningPrinted4token_[0] = true;
+    }
+    return;
+  }
 
   edm::Handle<reco::MuonCollection> muoHandle;
   iEvent.getByToken( muoToken_, muoHandle );
-
+  if ( !muoHandle.isValid() ) {
+    if (!warningPrinted4token_[1]) {
+      edm::LogWarning("BPHMonitor") << "skipping events because the collection " << muoInputTag_.label().c_str() << " is not available";
+      warningPrinted4token_[1] = true;
+    }
+    return;
+  }
 
   edm::Handle<reco::TrackCollection> trHandle;
   iEvent.getByToken( trToken_, trHandle );
+  if ( !trHandle.isValid() ) {
+    if (!warningPrinted4token_[2]) {
+      edm::LogWarning("BPHMonitor") << "skipping events because the collection " << trInputTag_.label().c_str() << " is not available";
+      warningPrinted4token_[2] = true;
+    }
+    return;
+  }
+  if ( !trHandle.isValid() ) {
```

Comment:
@mtosi why a second time? Is it a double cut-and-paste? I am missing the logic here...

Reply:
:(